crates/contracts/hydentity/src/main.rs

```diff
@@ -3,14 +3,15 @@
 
 extern crate alloc;
 
-use hydentity::execute;
+use hydentity::{execute, Hydentity};
 use sdk::guest::{commit, GuestEnv, Risc0Env};
-use sdk::ContractInput;
 
 risc0_zkvm::guest::entry!(main);
 
 fn main() {
     let env = Risc0Env {};
-    let input: ContractInput = env.read();
-    commit(env, input.clone(), execute(input));
+    let (hydentity_initial_state, contract_input) = env.read::<Hydentity>();
```

Comment:
possible to have it named state ou initial_state ? De manière à avoir au maximum le même `main` dans tous les contrats ?

En fait, j'aimerais que ce fichier devienne tellement tout le temps le même, que ce soit juste pointless de venir le regarder. 

et ça permet de copier coller le fichier, n'avoir vraiment "que" le type à changer :) 

bon, ok, c'est du maxi détail :DD